### ISSUE TITLE
Identify loop variable overriding

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#657]
It should identify if the variable name used on loopings (both `$key` or `$value`) conflicts with a previous defined variable. It can cause an accidental variable override.

```php
$values = [ 1, 2, 3 ];
$value  = 123;

foreach ($values as $value) {
    echo $value;
}

// Should be the "original" value (123).
// But it was overrided accidentally on looping.
echo $value;
```

The `$value` variable on `foreach` looping should warn something like "_this variable name was declared outside this looping and will be overrided here_".
